packages/static-renderer/src/helpers.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ExtensionAttribute, mergeAttributes } from '@tiptap/core'
+
+import type { MarkType, NodeType } from './types'
+
+/**
+ * This function returns the attributes of a node or mark that are defined by the given extension attributes.
+ * @param nodeOrMark The node or mark to get the attributes from
+ * @param extensionAttributes The extension attributes to use
+ * @param onlyRenderedAttributes If true, only attributes that are rendered in the HTML are returned
+ */
+export function getAttributes(
+  nodeOrMark: NodeType | MarkType,
+  extensionAttributes: ExtensionAttribute[],
+  onlyRenderedAttributes?: boolean,
+): Record<string, any> {
+  const nodeOrMarkAttributes = nodeOrMark.attrs
+
+  if (!nodeOrMarkAttributes) {
+    return {}
+  }
+
+  return extensionAttributes
+    .filter(item => {
+      if (item.type !== nodeOrMark.type) {
```

Comment:
on my testing, I had to use `nodeOrMark.type.name`, since otherwise it would not match anything.

Reply:
Ah, yea. Since this is shared between JSON and prosemirror objects, I added a check to see if it is a string or not first